minimint/src/consensus/mod.rs

```diff
@@ -8,42 +8,54 @@ use crate::config::ServerConfig;
 use crate::consensus::conflictfilter::ConflictFilterable;
 use crate::consensus::interconnect::MinimintInterconnect;
 use crate::db::{
-    AcceptedTransactionKey, DropPeerKey, DropPeerKeyPrefix, ProposedTransactionKey,
-    ProposedTransactionKeyPrefix, RejectedTransactionKey,
+    AcceptedTransactionKey, DropPeerKey, DropPeerKeyPrefix, EpochHistoryKey, LastEpochKey,
+    ProposedTransactionKey, ProposedTransactionKeyPrefix, RejectedTransactionKey,
 };
 use crate::outcome::OutputOutcome;
 use crate::rng::RngGenerator;
 use crate::transaction::{Input, Output, Transaction, TransactionError};
 use futures::future::select_all;
 use hbbft::honey_badger::Batch;
-use minimint_api::db::batch::{BatchTx, DbBatch};
+use minimint_api::db::batch::{AccumulatorTx, BatchItem, BatchTx, DbBatch};
 use minimint_api::db::Database;
 use minimint_api::encoding::{Decodable, Encodable};
 use minimint_api::module::audit::Audit;
 use minimint_api::{FederationModule, OutPoint, PeerId, TransactionId};
+use minimint_core::epoch::*;
 use minimint_core::modules::ln::{LightningModule, LightningModuleError};
 use minimint_core::modules::mint::{Mint, MintError};
 use minimint_core::modules::wallet::{Wallet, WalletError};
 use minimint_core::outcome::TransactionStatus;
-use minimint_derive::UnzipConsensus;
 use rand::{CryptoRng, RngCore};
 use serde::{Deserialize, Serialize};
-use std::collections::HashSet;
+use std::collections::{BTreeMap, HashSet};
+use std::iter::FromIterator;
 use std::sync::Arc;
 use thiserror::Error;
 use tokio::sync::Notify;
 use tracing::{debug, error, info, info_span, instrument, trace, warn};
 
-#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize, UnzipConsensus)]
-pub enum ConsensusItem {
-    Transaction(Transaction),
-    Mint(<Mint as FederationModule>::ConsensusItem),
-    Wallet(<Wallet as FederationModule>::ConsensusItem),
-    LN(<LightningModule as FederationModule>::ConsensusItem),
+pub type ConsensusOutcome = Batch<Vec<ConsensusItem>, PeerId>;
+pub type HoneyBadgerMessage = hbbft::honey_badger::Message<PeerId>;
+
+// TODO remove HBBFT `Batch` from `ConsensusOutcome`
+#[derive(Debug, Clone)]
+pub struct ConsensusOutcomeConversion(pub ConsensusOutcome);
```

Comment:
I wonder if it would have been easier to define our own `ToConsensusOutcome` trait instead.

Reply:
(can be fixed in follow-up)